cluster-autoscaler/cloudprovider/gce/cache.go

```diff
@@ -222,9 +224,28 @@ func (gc *GceCache) getMigNoLock(migRef GceRef) (mig Mig, found bool) {
 
 // RegenerateInstanceCacheForMig triggers instances cache regeneration for single MIG under lock.
 func (gc *GceCache) RegenerateInstanceCacheForMig(migRef GceRef) error {
+	klog.V(4).Infof("Regenerating MIG information for %s", migRef.String())
+
+	instances, err := gc.GceService.FetchMigInstances(migRef)
+	if err != nil {
+		klog.V(4).Infof("Failed MIG info request for %s: %v", migRef.String(), err)
+		return err
+	}
+
 	gc.cacheMutex.Lock()
 	defer gc.cacheMutex.Unlock()
-	return gc.regenerateInstanceCacheForMigNoLock(migRef)
+
+	// cleanup old entries
+	gc.removeInstancesForMigs(migRef)
```

Comment:
this code(239-248) is shared with regenerateInstanceCacheForMigNoLock. Consider moving it to one func used that would be used in both funcs.